src/Language/PureScript/Sugar/TypeClasses.hs

```diff
@@ -229,8 +229,29 @@ typeClassMemberToDictionaryAccessor _ _ _ _ = error "Invalid declaration in type
 unit :: Type
 unit = TypeApp tyObject REmpty
 
+fillDerivingDecls :: (MonadError MultipleErrors m, Applicative m) => Qualified ProperName -> [Type] -> [Declaration] -> [Declaration] -> Desugar m [Declaration]
+fillDerivingDecls className tys tyDecls decls = do
+  let missingDecls = mapMaybe declName tyDecls \\ mapMaybe declName decls
+  let mkMissingDecl cls fun =
+          case tys of
+            [TypeConstructor t] | unQualify className == ProperName cls && Ident fun `elem` missingDecls -> return [ValueDeclaration (Ident fun) Public [] (Right $ TypeClassInstanceMemberFunction (Ident fun) className t)]
+            _ -> return []
+  extraDecls <- concat <$> mapM (uncurry mkMissingDecl)
+                  [("Generic","toSpine")
```

Comment:
Typically, function names out of the Prelude go into the `Constants` module. Maybe these should go there too.